src/draw_target.rs

```diff
@@ -214,15 +222,15 @@ enum TargetKind {
 }
 
 impl TargetKind {
-    fn last_line_count(&mut self) -> Option<&mut usize> {
+    fn adjust_last_line_count(&mut self, num: isize) {
```

Comment:
This feels overly generic. IIRC we don't actually need to adjust in both directions so this could just take a `usize` (and adjust the method name accordingly -- `adjust_last_line_count` also seems overly verbose to me), which would get rid of most of the casting (and we can use a `saturating_sub()`).

Reply:
Yeah, I think the `Clear` vs `Keep` enum is the least ugly.

Reply:
@chris-laplante I think this just needs the enum change and we should get 0.17.1 out there. Can you make that change or shall I take care of it?